doc-cheat-sheet.rst

```diff
@@ -0,0 +1,611 @@
+
+Headings
+========
+
+.. list-table::
+   :header-rows: 1
+
+   * - Input
+     - Description
+   * - .. code::
+
+          Title
+          =====
+     - Page title and H1 heading
+   * - .. code::
+
+          Heading
+          -------
+     - H2 heading
+   * - .. code::
+
+          Heading
+          ^^^^^^^
+     - H3 heading
+   * - Underlines with other characters, for example, ``"``, ``~``, ``#`` or ``*``
+     - Further headings
+
+Underlines must be at least as long as the title or heading.
+
+Adhere to the following conventions:
+
+- Do not use consecutive headings without intervening text.
+- Do not skip levels (for example, do not follow an H2 heading with an H4 heading).
+- Use sentence style for headings (capitalise only the first word).
+
+Inline formatting
+=================
+
+.. list-table::
+   :header-rows: 1
+
+   * - Input
+     - Output
+   * - ``:guilabel:`UI element```
+     - :guilabel:`UI element`
+   * - ````code````
+     - ``code``
+   * - ``:file:`file path```
+     - :file:`file path`
+   * - ``:command:`command```
+     - :command:`command`
+   * - ``*Italic*``
+     - *Italic*
+   * - ``**Bold**``
+     - **Bold**
+
+Adhere to the following conventions:
+
+- Use italics sparingly. Common uses for italics are titles and names (for example, when referring to a section title that you cannot link to, or when introducing the name for a concept).
+- Use bold sparingly. Avoid using bold for emphasis and rather rewrite the sentence to get your point across.
+
+Code blocks
+===========
+
+To start a code block, either end the introductory paragraph with two colons (``::``) and indent the following code block, or explicitly start a code block with ``.. code::``.
+In both cases, the code block must be surrounded by empty lines.
+
+When explicitly starting a code block, you can specify the code language to enforce a specific lexer, but in many cases, the default lexer works just fine.
+
+
+.. list-table::
+   :header-rows: 1
+
+   * - Input
+     - Output
+   * - .. code::
+
+          Demonstrate a code block::
+
+            code:
+             - example: true
+     - Demonstrate a code block::
+
+         code:
+         - example: true
+   * - .. code::
+
+          .. code::
+
+             # Demonstrate a code block
+             code:
+             - example: true
+     - .. code::
+
+          # Demonstrate a code block
+          code:
+          - example: true
+   * - .. code::
+
+          .. code:: yaml
+
+             # Demonstrate a code block
+             code:
+             - example: true
+     - .. code:: yaml
+
+          # Demonstrate a code block
+          code:
+          - example: true
+
+Links
+=====
+
+How to link depends on if you are linking to an external URL or to another page in the documentation.
+
+External links
+--------------
+
+For external links, use the following syntax:
```

Comment:
For OpenStack, the style guide says to define the links at the bottom of the page.

Personally, I see no benefit in that. It just makes it harder to add a link, but if you need to update links, you still need to go through all files.
What I think is beneficial is having a separate file for defining ALL links, which is then included automatically in every content file, so that all links are available in all pages. That means that a broken link can easily be fixed everywhere by just changing it in the links file.

This does add some overhead when editing though, that's why I didn't include it here.

Reply:
I agree it does take more time to create links but once it's done it makes legibility of a source page very much better.

The idea about a central link page is alluring.

Reply:
That's what I've done before. Defining common links in a separate document and using `.. include:: /links.txt` to refer to them. External links specific to a page are defined in that page.

Reply:
I've added both options (using directly and defining at the bottom of the page) now - we don't need to be too restrictive here.

And I'll create another PR for adding a common link file.